x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_ml_inference_button.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiButton, EuiToolTip } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { KibanaLogic } from '../../../../../shared/kibana/kibana_logic';
+
+export interface AddMLInferencePipelineButtonProps {
+  onClick: () => void;
+}
+export const AddMLInferencePipelineButton: React.FC<AddMLInferencePipelineButtonProps> = ({
+  onClick,
+}) => {
+  const { capabilities } = useValues(KibanaLogic);
+  const hasMLPermissions = capabilities?.ml?.canAccessML ?? false;
```

Comment:
I'm not 100% sure if this is ok to use or not, but it seemed better than making the permissions call directly when there is already the `capabilities` object with a lot of ml permission available on it.

Reply:
I'm not entirely certain either - but, if it works, great! If not, we know the fallback to use the API